pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+  instances to compile correctly, the corresponding classes must be in the
+  repository.
+
+  - compile_mof adds CIM objects to the repository from mof definitions
+  contained in mof files. For classes to compile correctly, the qualifier
+  declarations for the target classes must exist in the repository.  For
+  instances to compile correctly, the corresponding classes must be in the
+  repository.
+
+FakeWBEMConnection provides responses for the WBEMConnection methods based on
+the data in the repository including responding with the correct CIMError
+exceptions when error conditions are found.
+
+The repository allows use of multiple namespaces with the default being the
+standard pywmbe default namespace.
+
+This mocker uses the same APIs as the standard WBEMConnections (it actually
+uses the methods and only mocks the communication with the WBEMServer) and
+returns syntatically the same forms of data and exceptions.
+
+- Class methods:
+
+The methods that get data require the expected classes to be in the repository
+before the mock call. The methods are:
+
+  - GetClass - Gets a single class from a namespace in the repository
+  - EnumerateClasses - Gets list of classes from a namespace in the
+    repository
+  - EnumerateClassNames - Gets list of class names from the repository
+  - CreateClass - Creates a class in the repository
+  - DeleteClass - Deletes a class in the repository
+
+-CIMQualifierDeclaration methods:
+
+  - SetQualifier - Sets a new QualifierDeclaration into a namespace in the
+    repository
+  - GetQualifier - Gets a QualifierDeclaration from the repository
+  - EnumerateQualifiers - Gets all QualifierDeclaration entities from a
+    namespace in the repository
+
+-CIMInstance methods
+
+The methods that get data require both classes and instances in the repository
+for the instances to be recovered.
+
+  - GetInstance - Gets an instance from a namespace in the repository
+  - EnumerateInstances - Gets instances for a class defined in a namespace
+    in the repository.  If there is no class repository, it gets instances
+    for the defined class.  If a class definition for the target class is in
+    the repository, it attempts to get instances of this class and all
+    subclasses
+  - EnumerateInstanceNames - Gets instance names for a class defined in
+    a namespace in the repository.  If there is no class repository, it gets
+    names or the defined class.  If a class definition for the target class is
+    in the repository, it attempts to get instances of this class and all
+    subclasses
+  - CreateInstance - TODO
+  - ModifyInstance - TODO
+  - ExecQuery - This method will return fixed data based on the query string.
+    It does not execute the select statement itself nor does it parse it.
+    An incorrect select statement will be ignored.
+
+- Associators and References
+ - AssociatorNames
+ - Associators
+ - ReferenceNames
+ - References
+
+Supports both class and instance methods
+
+-InvokeMethod
```

Comment:
For InvokeMethod, we should say that the user can establish a function to be called.

Reply:
Done but left a TODO for us to refine.